language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -193,16 +193,16 @@ dummyKind = Type ()
 dummyType :: Type TyNameWithKind ()
 dummyType = TyVar () dummyTyName
 
--- | Extract type of a term.
-typeOf :: Term TyNameWithKind NameWithType a -> TypeCheckM a (Type TyNameWithKind ())
-typeOf (Var _ (NameWithType (Name (_, ty) _ _))) = pure (void ty)
-typeOf (LamAbs _ _ ty t)                         = TyFun () (void ty) <$> typeOf t
+-- | Extract type of a term. The resulting type is normalized.
+typeOf :: Term TyNameWithKind NameWithType a -> TypeCheckM a (NormalizedType TyNameWithKind ())
```

Comment:
I think the type-level `a` in this module means sometimes "annotation" and sometimes "tyname". This is confusing.

Reply:
Agreed, but I'm not going to change that now.